File: haskell/src/Controllers/LoginController.hs
{-# LANGUAGE OverloadedStrings #-}

module Controllers.LoginController where
import Util.Entry as Entry
import Util.Menu as Menu
import qualified System.IO.Streams as Streams
import Data.Text as TS
import Util.Util as Util
import Database.ConnectionDB as ConnectionDB
import Database.MySQL.Base
import System.Console.ANSI

signIn :: (String -> IO ()) -> IO() 
signIn originalMenu = do 
    clearScreen
    putStrLn("Digite o usuário")
    user <- Entry.lerEntrada
    putStrLn("Digite a senha")
    password <- Entry.lerEntrada
    (defs, is) <- getUser user password
    matriz <- Streams.toList is

    if matriz /= []
        then do
            let nome = Util.convert(Util.getName (Util.matrizToList matriz))
            clearScreen
            menu originalMenu matriz
    else
        originalMenu("Usuário não encontrado!")

signUp :: (String -> IO ()) -> IO()
signUp originalMenu = do
    clearScreen 
    putStrLn("Nome: ")
    nome <- Entry.lerEntrada
    putStrLn("Profissao: ")
    profissao <- Entry.lerEntrada
    putStrLn("CPF: ")
    cpf <- Entry.lerEntrada
    putStrLn("Matricula: ")
    matricula <- Entry.lerEntrada
    putStrLn("Usuario: ")
    usuario <- Entry.lerEntrada
    putStrLn("Senha: ")
    senha <- Entry.lerEntrada

    (defs, is) <- verifyData cpf matricula usuario
    matriz <- Streams.toList is
    
    validateRegistry originalMenu matriz nome cpf usuario senha matricula profissao

File: prolog/src/Controllers/loginController.pl
:- include('../Database/connectionDB.pl').
:- include('../Util/menu.pl').

signIn :- 
    tty_clear,
    write("Digite o usuário"), nl,
    read(Usuario),
    write("Digite a senha"), nl,
    read(Senha),
    (get_user(Usuario, Senha, Result) -> menu(Result); write("Usuário não encontrado!")).

signUp :-
    write('Digite seu nome: '), nl,
    read(Nome),
    write('Digite seu cpf: '), nl,
    read(Cpf),
    write('Digite sua Matricula: '), nl,
    read(Matricula),
    write('Digite sua Profissão: '), nl,
    read(Profissao),
    write('Digite seu Usuario: '), nl,
    read(Usuario),
    write('Digite sua Senha: '), nl,
    read(Senha),
    (validate_registry(Cpf, Matricula, Usuario, Result) -> write('Usuario já possui uma conta no sistema!'),nl,write("digite E. para continuar..."), nl,
    read(buffer), menu(Result);
    cadastra_usuario(Nome, Cpf, Matricula, Profissao, Usuario, Senha, F) -> write('Cadastro Realizado!'), nl, write("digite E. para continuar..."), nl,
    read(buffer), get_user(Usuario, Senha, R), menu(R);
    write('Não foi possivel realizar o cadastro'), nl, write("digite E. para continuar..."), nl,
    read(buffer), menu(Result)).